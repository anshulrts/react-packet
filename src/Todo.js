import React, { Component } from 'react';
import './Todo.css';

class Todo extends Component {
    render() {
        return <div className="Todo">{ this.props.description }</div>;
    }
}

export default Todo;