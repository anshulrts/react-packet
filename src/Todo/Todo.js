import React, { Component } from 'react';
import styles from  './Todo.module.scss';

class Todo extends Component {

    constructor(props) {
        super(props);
        this.state = {
            description: this.props.description,
            done: false,
            critical: false
        }

        this.markAsDone = this.markAsDone.bind(this);
        this.removeTodo = this.removeTodo.bind(this);
        this.markCritical = this.markCritical.bind(this);
    }

    markAsDone() {
        this.setState({ done: true });
    }

    cssClasses() {
        let classes = [];

        if(this.state.critical) {
            classes = [styles.critical];
        }
        else {
            classes = [styles.todo];
        }
        if(this.state.done) {
            classes = [...classes, styles.done];
        }
        return classes.join(' ');
    }

    removeTodo() {
        this.props.removeTodo(this.state.description);
    }

    markCritical() {
        this.setState({ critical: true });
    }

    render() {
        return (
            <div className={this.cssClasses()}>
                { this.state.description }
            <br />
            <hr className={styles.redDivider} />
            <button className="MarkDone" onClick={ this.markAsDone }>Mark as Done</button>
            <button className="RemoveTodo" onClick={ this.removeTodo }>Remove Me</button>
            <button className="MarkCritical" onClick={ this.markCritical }>Mark as Critical</button>
            </div>
        );
    }
}

export default Todo;