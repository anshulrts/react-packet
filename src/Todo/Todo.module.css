.todo {
    border: 2px solid black;
    text-align: center;
    background: #f5f5f5;
    color: #333333;
    margin: 20px;
    padding: 20px;
}

.done {
    background: #f58888;
  }

  .redDivider {
    border: 2px solid red;
  }