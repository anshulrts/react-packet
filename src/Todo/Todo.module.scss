@import "../shared";

.todo {
    border: 2px solid black;
    text-align: center;
    background: $todo-normal;
    color: #333333;
    margin: 20px;
    padding: 20px;
}

.critical {
    composes: todo;
    background: $todo-critical;
}

.done {
    background: $todo-complete;
  }

  .redDivider {
    border: 2px solid red;
  }