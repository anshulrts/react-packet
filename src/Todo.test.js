import React, { Component } from 'react';
import ReactDOM from "react-dom";
import  { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Todo from './Todo';

const description = "New Todo";
const mockRemoveTodo = jest.fn();
const component = shallow(
    <Todo description={description} removeTodo={mockRemoveTodo} />
);

describe(Todo, () => {
    it("renders wihtout crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Todo />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders and matches our snapshot", () => {
        const component = renderer.create(<Todo description="Yo" />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});