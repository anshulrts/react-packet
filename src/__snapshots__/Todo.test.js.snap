// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders and matches our snapshot 1`] = `
<div
  className="Todo"
>
  Yo
  <br />
  <button
    onClick={[Function]}
  >
    Mark as Done
  </button>
  <button
    onClick={[Function]}
  >
    Remove Me
  </button>
</div>
`;
